src/Xamarin.Android.Build.Utilities/Sdks/MonoDroidSdkUnix.cs

```diff
@@ -22,6 +22,9 @@ class MonoDroidSdkUnix : MonoDroidSdkBase
 		protected override string FindRuntime ()
 		{
 			string monoAndroidPath = Environment.GetEnvironmentVariable ("MONO_ANDROID_PATH");
+			if (monoAndroidPath == null)
+				// for Mono.Posix and Mono.Data.Sqlite builds in xamarin-android.
+				monoAndroidPath = Path.GetFullPath (Path.Combine (new Uri (GetType ().Assembly.CodeBase).LocalPath, "..", "..", "..", "..", ".."));
```

Comment:
if MONO_ANDROID_PATH is a valid value (via the xabuild script) the code will exit [1] so in that case your code path will not be hit and neither would mine. We already have code in place to search alternate directories, I'm just suggesting we use it :) e.g

```
var additionalSearchPaths = new [] {
  Path.GetFullPath (Path.Combine (new Uri (GetType ().Assembly.CodeBase).LocalPath, "..", "..", "..", "..", "..", "lib", "mandroid")),
  Path.Combine (personal, @".xamarin.android/lib/mandroid"),
};
```



[1] https://github.com/atsushieno/xamarin-android/blob/8db545653004741711f39d75fa598e1f5a88ace6/src/Xamarin.Android.Build.Utilities/Sdks/MonoDroidSdkUnix.cs#L32